app/models/papi_client.rb

```diff
@@ -305,41 +306,58 @@ def create_virtual_machine_object(machine_type: DEFAULT_MACHINE_TYPE,
   def get_command_line(study_file:, action:, user_metrics_uuid:, params_object: nil)
     validate_action_by_file(action, study_file)
     study = study_file.study
-    command_line = "python ingest_pipeline.py --study-id #{study.id} --study-file-id #{study_file.id} " \
-                   "--user-metrics-uuid #{user_metrics_uuid} #{action}"
-
+    # Docker accepts command line in array form for better tokenization of parameters
+    command_line = [
+      'python', 'ingest_pipeline.py', '--study-id', study.id.to_s, '--study-file-id', study_file.id.to_s,
+      '--user-metrics-uuid', user_metrics_uuid, action.to_s
+    ]
+    action_cli_opt = Parameterizable.to_cli_opt(action)
     case action.to_s
     when 'ingest_expression'
-      if study_file.file_type == 'Expression Matrix'
-        command_line += " --matrix-file #{study_file.gs_url} --matrix-file-type dense"
-      elsif study_file.file_type === 'MM Coordinate Matrix'
+      case study_file.file_type
+      when 'Expression Matrix'
+        command_line += ['--matrix-file', study_file.gs_url, '--matrix-file-type', 'dense']
+      when 'MM Coordinate Matrix'
         bundled_files = study_file.bundled_files
-        genes_file = bundled_files.detect {|f| f.file_type == '10X Genes File'}
-        barcodes_file = bundled_files.detect {|f| f.file_type == '10X Barcodes File'}
-        command_line += " --matrix-file #{study_file.gs_url} --matrix-file-type mtx" \
-                      " --gene-file #{genes_file.gs_url} --barcode-file #{barcodes_file.gs_url}"
+        genes_file = bundled_files.detect { |f| f.file_type == '10X Genes File' }
+        barcodes_file = bundled_files.detect { |f| f.file_type == '10X Barcodes File' }
+        command_line += [
+          '--matrix-file', study_file.gs_url, '--matrix-file-type', 'mtx', '--gene-file', genes_file.gs_url,
+          '--barcode-file', barcodes_file.gs_url
+        ]
```

Comment:
Nice, this has notably better visual signal-to-noise.

Reply:
[785e2a4](https://github.com/broadinstitute/single_cell_portal_core/pull/1758/commits/785e2a4069db000fc9de1428abd2905ad97c2d3a)